packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -204,6 +232,25 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
             items: action.type === 'loading' ? [] : data.items,
             abortController: action.abortController
           };
+        case 'filtering':
```

Comment:
Could this case be combined with the above `loading` and `sorting` cases? I think the current state value that is set above is wrong though, it should be `action.type` not always `loading`.

Reply:
Ahh, yeah I think I can combine them not sure why I didn't, good catch